go/chat/unfurl/scraper_test.go

```diff
@@ -0,0 +1,124 @@
+package unfurl
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"path/filepath"
+	"testing"
+
+	"github.com/keybase/client/go/logger"
+	"github.com/keybase/client/go/protocol/chat1"
+	"github.com/stretchr/testify/require"
+)
+
+type dummyHTTPSrv struct {
+	t   *testing.T
+	srv *http.Server
+}
+
+func newDummyHTTPSrv(t *testing.T) *dummyHTTPSrv {
+	return &dummyHTTPSrv{
+		t: t,
+	}
+}
+
+func (d *dummyHTTPSrv) Start() string {
+	localhost := "127.0.0.1"
+	listener, err := net.Listen("tcp", fmt.Sprintf("%s:0", localhost))
+	require.NoError(d.t, err)
+	port := listener.Addr().(*net.TCPAddr).Port
+	d.srv = &http.Server{
+		Addr: fmt.Sprintf("%s:%d", localhost, port),
+	}
+	http.HandleFunc("/", d.handle)
+	go d.srv.Serve(listener)
+	return d.srv.Addr
+}
+
+func (d *dummyHTTPSrv) Stop() {
+	require.NoError(d.t, d.srv.Close())
+}
+
+func (d *dummyHTTPSrv) handle(w http.ResponseWriter, r *http.Request) {
+	w.WriteHeader(200)
+	name := r.URL.Query().Get("name")
+	dat, err := ioutil.ReadFile(filepath.Join("testcases", name))
+	require.NoError(d.t, err)
+	_, err = io.Copy(w, bytes.NewBuffer(dat))
+	require.NoError(d.t, err)
+}
+
+func strPtr(s string) *string {
+	return &s
+}
+
+func TestScraper(t *testing.T) {
+	scraper := NewScraper(logger.NewTestLogger(t))
+	srv := newDummyHTTPSrv(t)
+	addr := srv.Start()
+	defer srv.Stop()
+	testCase := func(name string, expected chat1.UnfurlRaw) {
+		res, err := scraper.Scrape(context.TODO(), "http://"+addr+"/?name="+name)
+		require.NoError(t, err)
+		etyp, err := expected.UnfurlType()
+		require.NoError(t, err)
+		rtyp, err := res.UnfurlType()
+		require.NoError(t, err)
+		require.Equal(t, etyp, rtyp)
+		t.Logf("expected: %v res: %v", expected, res)
+		switch rtyp {
+		case chat1.UnfurlType_GENERIC:
+			e := expected.Generic()
+			r := res.Generic()
+			require.Equal(t, e.Title, r.Title)
+			require.Equal(t, e.SiteName, r.SiteName)
+			require.True(t, (e.Description == nil && r.Description == nil) || (e.Description != nil && r.Description != nil))
+			require.True(t, e.Description == nil || *e.Description == *r.Description)
+			require.True(t, (e.ImageUrl == nil && r.ImageUrl == nil) || (e.ImageUrl != nil && r.ImageUrl != nil))
+			require.True(t, e.ImageUrl == nil || *e.ImageUrl == *r.ImageUrl)
+			require.True(t, (e.FaviconUrl == nil && r.FaviconUrl == nil) || (e.FaviconUrl != nil && r.FaviconUrl != nil))
+			require.True(t, e.FaviconUrl == nil || *e.FaviconUrl == *r.FaviconUrl)
+		default:
+			require.Fail(t, "unknown unfurl typ")
+		}
+	}
+	testCase("wsj0", chat1.NewUnfurlRawWithGeneric(chat1.UnfurlGenericRaw{
```

Comment:
can you add `.html` to all these files? just helps with default rendering on github/editors